tools/Google.Cloud.Tools.ReleaseManager.Tests/CheckVersionCompatibilityCommandTest.cs

```diff
@@ -0,0 +1,115 @@
+        public static TheoryData<TestCase> ReleasePrTestData { get; } = new TheoryData<TestCase>
+        {
+            // The first constructor argument is the new version.
+            // The remaining constructor arguments are tagged versions.
+            // The collection initializer is used to add expected comparisons.
+
+            // First release: no comparisons
+            new TestCase("1.0.0-beta01"),
+            // Straight to 1.0.0 is pretty bold, but if we don't have any previous releases, we don't have anything to compare it to.
+            new TestCase("1.0.0"),
+            // Beta to first stable release
+            // We perform the comparison with the beta even though we allow anything to change. It's still useful to see those changes.
+            new TestCase("1.0.0", "1.0.0-beta01") { { "1.0.0-beta01", Level.Major } },
+
+            // First beta of a new minor
+            new TestCase("1.1.0-beta01", "1.0.0-beta01", "1.0.0") { { "1.0.0", Level.Minor } },
+
+            // Second beta of a new minor
+            new TestCase("1.1.0-beta02", "1.0.0-beta01", "1.0.0", "1.1.0-beta01")
+            {
+                { "1.0.0", Level.Minor },       // We're aiming for 1.1.0, and there can't be breaking changes there
+                { "1.1.0-beta01", Level.Major } // We can break beta-only features, but we'd like to know that.
+            },
+
+            // Release of a new minor
+            new TestCase("1.1.0", "1.0.0-beta01", "1.0.0", "1.1.0-beta01")
+            {
+                { "1.0.0", Level.Minor },       // We're aiming for 1.1.0, and there can't be breaking changes there
+                { "1.1.0-beta01", Level.Major } // We can break beta-only features, but we'd like to know that.
+            },
+
+            // First beta of a new major
+            new TestCase("2.0.0-beta01", "1.0.0", "1.1.0") // Any beta releases in 1.x are irrelevant now
+            {
+                { "1.1.0", Level.Major }
+            },
+
+            // Second beta of a new major
+            new TestCase("2.0.0-beta02", "1.0.0-beta01", "1.0.0", "1.1.0-beta01", "2.0.0-beta01")
+            {
+                { "1.1.0", Level.Major },        // So we know what we've broken compared with last GA
+                { "2.0.0-beta01", Level.Major }, // So we know what we've broken compared with last beta
+            },
+
+            // First release of a new major
+            new TestCase("2.0.0", "1.0.0-beta01", "1.0.0", "1.1.0-beta01", "2.0.0-beta01")
+            {
+                { "1.1.0", Level.Major },        // So we know what we've broken compared with last GA
+                { "2.0.0-beta01", Level.Major }, // So we know what we've broken compared with last beta
+            }
+        };
+
+        // Non-release PRs will have a version that is already tagged (and published)
+        public static TheoryData<TestCase> NonReleasePrTestData = new TheoryData<TestCase>
+        {
+            // TODO
```

Comment:
Not sure if we could add a hint as to what the intended version is or if the code can run assuming the same version as increment-version would generate. We can always use allow breaking changes to bump major, but we'll have to continue to use it until we have released the first major beta.

Reply:
I think if we know we'll do a major next, we can go to 2.0.0-beta00. Which reminds me, I don't mention that anywhere here...

Reply:
Ah yes makes sense.